labs/lab9/9.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Lab 9 - Web Application
+layout: lab
+---
+
+## Overview
+
+We’re going to set up a flask app. And we are going to call it basicapp
+
+There is 3 main component to this lab. We will install Nginx, uWSGI, and flask through the apt command. Then we will git clone an example app and config files from class repository. Next, we will modify the configuration files. If configured correctly, we'll go through the startup procedure. Then end it with some debugging tips and tricks. 
+
+## Note
+
+* No sudo was included in any command examples. Use your best judgment to include sudo or not. 
+
+### The Setup
+
+We need virtualenv and git for staging. Run
+
+    apt install virtualenv git
+
+Next, we'll create the directory where our project will be located. We'll just call it [basicapp] and we'll place it in /opt/. 
+
+    mkdir /opt/basicapp
+
+Next, we'll set up our virturalenv. Virtualenv is a python tool that lets you create a custom python execution environment that is specific to your project. This is so dependencies or modified libraries does not pollute your main coding environment. Let's create the virtualenv for this project and activate it. 
+
+    chown $USER:www-data basicapp
+    cd /opt/basicapp
+    virtualenv venv
+    source venv/bin/activate
+    
+
+You can see a (venv) in the beginning of your prompt. Now all your python code will execute the venv version of python. Located in "/opt/basicapp/venv/bin/python". And it'll add venv to top of your python path, that means your python library will search venv first before search global library directory. 
```

Comment:
you may need to explain what it means to "add venv to the top of your python path, ... python library will search venv first" - there's also grammar to correct in this paragraph and others.